packages/flutter/lib/src/cupertino/page.dart

```diff
@@ -0,0 +1,173 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+const double _kMinFlingVelocity = 1.0;  // screen width per second.
+const Color _kBackgroundColor = const Color(0xFFEFEFF4); // iOS 10 background color.
+
+/// Transitions used for standard iOS full page transitions by bringing coming in from the right on
+/// top of the previous screen and by becoming pushed off-screen to the left with a parallax effect
+/// below the next screen.
+class CupertinoPageTransition extends AnimatedWidget {
+  CupertinoPageTransition({
+    Key key,
+    // Linear route animation from 0.0 to 1.0 when this screen is being pushed.
+    @required Animation<double> incomingRouteAnimation,
```

Comment:
i'm not sure incoming/outgoing really captures the semantics of these two animations, but i don't have a better suggestion. :-(